File: 프로그래머스/lv1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.cpp
#include <string>
#include <vector>

using namespace std;

int Euclid(int a, int b)
{
    int r = a % b;
    if (r == 0)
        return b;

    return Euclid(b, r);
}

int Lcm(int a, int b)
{
    return a * b / Euclid(a, b);
}

vector<int> solution(int n, int m)
{
    vector<int> answer;
    answer.push_back(Euclid(n, m));
    answer.push_back(Lcm(n, m));

    return answer;
}

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
class Solution {

    int lcm(int a, int b) {
        return a * b / gcd(a, b);
    }

    int gcd(int a, int b) {
        if (b == 0) {
            return a;
        }
        return gcd(b, a % b);
    }

    public int[] solution(int n, int m) {
        return new int[]{gcd(n, m), lcm(n, m)};
    }
}